src/bin/common/parse_command.ml

```diff
@@ -0,0 +1,1157 @@
+      "Enable substitution of variables bounds by Let. The default \
+       behavior is to only substitute variables that are bound to a \
+       constant, or that appear at most once." in
+    Arg.(value & flag & info ["inline-lets"] ~docs ~doc) in
+
+  let rewriting =
+    let doc = "Use rewriting instead of axiomatic approach." in
+    Arg.(value & flag & info ["rwt"; "rewriting"] ~docs ~doc) in
+
+  let term_like_pp =
+    let doc = "Output semantic values as terms." in
+    Arg.(value & flag & info ["term-like-pp"] ~docs ~doc) in
+
+  Term.(ret (const mk_term_opt $
+             disable_ites $ inline_lets $ rewriting $ term_like_pp
+            ))
+
+let parse_theory_opt =
+
+  let docs = s_theory in
+
+  let disable_adts =
+    let doc = "Disable Algebraic Datatypes theory." in
+    Arg.(value & flag & info ["disable-adts"] ~docs ~doc) in
+
+  let inequalities_plugin =
+    let doc =
+      "Use the given module to handle inequalities of linear arithmetic." in
+    Arg.(value & opt string (get_inequalities_plugin ()) &
+         info ["inequalities-plugin"] ~docs ~doc) in
+
+  let no_ac =
+    let doc = "Disable the AC theory of Associative and \
+               Commutative function symbols." in
+    Arg.(value & flag & info ["no-ac"] ~docs ~doc) in
+
+  let no_contracongru =
+    let doc = "Disable contracongru." in
+    Arg.(value & flag & info ["no-contracongru"] ~docs ~doc) in
+
+  let no_fm =
+    let doc = "Disable Fourier-Motzkin algorithm." in
+    Arg.(value & flag & info ["no-fm"] ~docs ~doc) in
+
+  let no_nla =
+    let doc = "Disable non-linear arithmetic reasoning (i.e. non-linear \
+               multplication, division and modulo on integers and rationals). \
+               Non-linear multiplication remains AC." in
+    Arg.(value & flag & info ["no-nla"] ~docs ~doc) in
+
+  let no_tcp =
+    let doc = "Deactivate BCP modulo theories." in
+    Arg.(value & flag & info ["no-tcp"] ~docs ~doc) in
+
+  let no_theory =
+    let doc = "Completely deactivate theory reasoning." in
+    Arg.(value & flag & info ["no-theory"] ~docs ~doc) in
+
+  let restricted =
+    let doc =
+      "Restrict set of decision procedures (equality, arithmetic and AC)." in
+    Arg.(value & flag & info ["restricted"] ~docs ~doc) in
+
+  let tighten_vars =
+    let doc = "Compute the best bounds for arithmetic variables." in
+    Arg.(value & flag & info ["tighten-vars"] ~docs ~doc) in
+
+  let use_fpa =
+    let doc = "Enable support for floating-point arithmetic." in
+    Arg.(value & flag & info ["use-fpa"] ~docs ~doc) in
+
+  Term.(ret (const mk_theory_opt $
+             disable_adts $ inequalities_plugin $ no_ac $ no_contracongru $
+             no_fm $ no_nla $ no_tcp $ no_theory $ restricted $
+             tighten_vars $ use_fpa
+            )
+       )
+
+let main =
+
+  let file =
+    let doc =
+      "Source file. Must be suffixed by $(i,.ae), \
+       ($(i,.mlw) and $(i,.why) are depreciated, \
+       $(i,.smt2) or $(i,.psmt2)." in
```

Comment:
This is wrong and somewhat not really clear with the depreciated extensions in the middle of the sentence.
As far as I can tell, the discussion about the extension should take place in the documentation for the input format option (which should be referenced here), which should explain what formats are recognized by the heuristics for detecting format based on the extension. But even then, I don't see a problem with an input file with any extension if the input format is explicitly specified on the command-line.

Reply:
There is some work to do about the documentation and this PR is not about it. I started answering I don't know to some of your remarks but I'll just mark them as resolved and end it here. This PR is about splitting the options lib from their parsing in the binary, not about making the options clearer or more accurate since it's another work, it takes a lot of time and I am not yet qualified to do it. Thanks for your review, I changed some things that needed to be changed but I won't do this work.

Reply:
I suggest you make a PR about all these changes since you know Alt-Ergo better than me and these problems precede my arrival to the project.

Reply:
If i read this correctly this PR can be merged. @Gbury could you open and issue with your comment on this PR regarding the lack of documentation of the options module please ?

Reply:
@OCamlPro-mattiasdrp can you correct the last issues with indentation and style please and when the CI will be green we can merge this PR.

Reply:
> If i read this correctly this PR can be merged. @Gbury could you open and issue with your comment on this PR regarding the lack of documentation of the options module please ?

Well, I still think that unless we take care of updating and documenting the code that is touched (in any way) in a PR (except for urgent changes that fix unsoundness bugs), we'll likely never take the time to do it, and it'll always stay the thing to do "when we have time", which is unlikely to happen reasonably soon...